kiosk/src/Components/GameList.tsx

```diff
@@ -1,133 +1,119 @@
 import React, { useEffect, useRef, useContext, useCallback } from "react";
-import { KioskState } from "../Types";
-import configData from "../config.json";
-import "../Kiosk.css";
 import { Swiper, SwiperSlide } from "swiper/react";
-import {
-    Swiper as SwiperClass,
-    EffectCoverflow,
-    Keyboard,
-    Pagination,
-} from "swiper";
+import { Swiper as SwiperClass, Pagination } from "swiper";
 import "swiper/css";
 import "swiper/css/keyboard";
 import GameSlide from "./GameSlide";
 import { playSoundEffect } from "../Services/SoundEffectService";
 import { AppStateContext } from "../State/AppStateContext";
 import { selectGameByIndex } from "../Transforms/selectGameByIndex";
 import { launchGame } from "../Transforms/launchGame";
-import { gamepadManager } from "../Services/GamepadManager";
-import { getHighScores, getSelectedGameIndex } from "../State";
+import {
+    getHighScores,
+    getSelectedGameIndex,
+    getSelectedGameId,
+} from "../State";
+import * as GamepadManager from "../Services/GamepadManager";
+import * as NavGrid from "../Services/NavGrid";
+import { useOnControlPress, useMakeNavigable } from "../Hooks";
 
-interface IProps {
-    addButtonSelected: boolean;
-    deleteButtonSelected: boolean;
-}
+interface IProps {}
 
-const GameList: React.FC<IProps> = ({
-    addButtonSelected,
-    deleteButtonSelected,
-}) => {
+const GameList: React.FC<IProps> = ({}) => {
     const { state: kiosk } = useContext(AppStateContext);
-    const buttonSelected = addButtonSelected || deleteButtonSelected;
     const localSwiper = useRef<SwiperClass>();
+    const [userInitiatedTransition, setUserInitiatedTransition] =
+        React.useState(false);
+    const [pageInited, setPageInited] = React.useState(false);
```

Comment:
Not quite sure what "inited" is.. Is this to check initialization ?

Reply:
This is me fighting with Swiper for who gets to decide which game is selected on component mount. I don't like it. This carousel is feature rich, but surprisingly difficult to control.